src/flags.rs

```diff
@@ -0,0 +1,3 @@
+// when this flag is set, the block generator serialization is allowed to
+// contain back-references
+pub const ALLOW_BACKREFS: u32 = 0x2000000;
```

Comment:
it's pretty scary to duplicate this constant with the same name and same value as in `chia_rs`. It could be mistaken for being the same flag.

I think it would be better to use a different name and different value, to make it clear that it's independent from the flag in `chia_rs`. It should also probably live in the wasm binding, since that's the only place it's used.

At some point we'll have type safe flags, and then this will be less of a concern.

Reply:
I agree that a more natural API would just be to expose both the "with backreferences" and "without backreferences" calls separately, since they're quite different, or maybe include a boolean flag. If you want to the API to resemble that `chia_rs` api, that's an argument for using a mask of flags.

Reply:
At the very least, move the flag and the branch to the `wasm` api.

Reply:
I'd like to keep the value `0x2000000` the same as `chia_rs` because if changing the value, wasm/JavaScript developers needs special code to make it compatible with other chia applications.
I think `clvm_rs` is more likely responsible for serialization (thus `ALLOW_BACKREFS` flag) rather than `chia_rs` so in my heart `ALLOW_BACKREFS` should be defined in `clvm_rs` and not in `chia_rs`...

But it's OK to hide the value to `wasm/` code and use only in the wasm.
I don't want to change the grand design and relationship between `chia_rs` and `clvm_rs` for now.